addons/website_sale/static/src/js/website_sale.editor.js

```diff
@@ -811,6 +823,58 @@ options.registry.WebsiteSaleProductPage = options.Class.extend({
         dialog.mount(document.body);
     },
 
+    async _convertAttachmentToWebp(attachment, imageEl) {
```

Comment:
I could not reproduce the file extension bug we discussed, so I think we can safely bet it was a bad fetch.
Everything seems to be working fine and apart from the extensions in capital letters you already fixed, I have no functional comment.

But since the code of `_convertAttachmentToWebp()` is derived and mostly copied/pasted from `onFileUploaded()` in `/web/static/src/views/fields/image/image_field.js`, you could maybe add a comment referring to it? So that if it ever needs to be modified, it's easily traceable and both are adapted at the same time. What do you think?